### PR TITLE
allow to watch signle file

### PR DESCRIPTION
**Pull request by vanyakosmos**
refactor and test PR #18

### CODE REVIEW

File: watchgod/watcher.py [hunk 1]
```diff
@@ -25,19 +25,29 @@ def should_watch_dir(self, entry):
...
         return True
 
-    def _walk(self, dir_path, changes, new_files):
+    def _walk(self, path, changes, new_files):
+        if os.path.isfile(path):
+            self._watch_file(path, changes, new_files)
+        else:
+            self._walk_dir(path, changes, new_files)
+
+    def _watch_file(self, path, changes, new_files, mtime=None):
```
Comment (samuelcolvin) [inline watchgod/watcher.py]: I'm not sure we need `mtime` argument here. the `mtime = os.stat(path).st_mtime` should always work and be no less performant.

File: watchgod/watcher.py [hunk 2]
```diff
@@ -25,19 +25,29 @@ def should_watch_dir(self, entry):
...
+
+    def _walk_dir(self, dir_path, changes, new_files):
         for entry in os.scandir(dir_path):
-            if entry.is_dir():
-                if self.should_watch_dir(entry):
-                    self._walk(entry.path, changes, new_files)
-            elif self.should_watch_file(entry):
+            if entry.is_dir() and self.should_watch_dir(entry):
+                self._walk_dir(entry.path, changes, new_files)
+            elif entry.is_file() and self.should_watch_file(entry):
```
Comment (samuelcolvin) [inline watchgod/watcher.py]: do we need `entry.is_file() and ` in here?

File: watchgod/watcher.py [hunk 3]
```diff
@@ -25,19 +25,29 @@ def should_watch_dir(self, entry):
...
+        elif old_mtime != mtime:
+            changes.add((Change.modified, path))
+
+    def _walk_dir(self, dir_path, changes, new_files):
         for entry in os.scandir(dir_path):
-            if entry.is_dir():
-                if self.should_watch_dir(entry):
-                    self._walk(entry.path, changes, new_files)
-            elif self.should_watch_file(entry):
+            if entry.is_dir() and self.should_watch_dir(entry):
```
Comment (samuelcolvin) [inline watchgod/watcher.py]: I think revert to the double if we had before, that way we don't call `elif entry.is_file() and self.should_watch_file(entry):` in the case that `entry` is a directory but we decide not to watch it.

File: watchgod/watcher.py (hunk 2 above)
Comment (vanyakosmos) [reply watchgod/watcher.py]: Not really, it just looks cleaner

Comment (samuelcolvin) [reply watchgod/watcher.py]: I think better to keep this stuff the way it was.

**Comment by samuelcolvin** [comment]
Amazing, thank you. Will try to deploy tomorrow.

**Pull request merged**